opensearch-operator/pkg/reconcilers/role_test.go

```diff
@@ -0,0 +1,585 @@
+				events = append(events, msg)
+			}
+			Expect(len(events)).To(Equal(1))
+			Expect(events[0]).To(Equal(fmt.Sprintf("Normal %s waiting for opensearch cluster to exist", opensearchPendingReason)))
+		})
+	})
+
+	When("cluster doesn't match status", func() {
+		BeforeEach(func() {
+			instance.Status.ManagedCluster = &opsterv1.OpensearchClusterSelector{
+				Name:      "somecluster",
+				Namespace: "somenamespace",
+			}
+			recorder = record.NewFakeRecorder(1)
+		})
+		It("should error", func() {
+			go func() {
+				defer GinkgoRecover()
+				defer close(recorder.Events)
+				_, err := reconciler.Reconcile()
+				Expect(err).To(HaveOccurred())
+			}()
+			var events []string
+			for msg := range recorder.Events {
+				events = append(events, msg)
+			}
+			Expect(len(events)).To(Equal(1))
+			Expect(events[0]).To(Equal(fmt.Sprintf("Warning %s cannot change the cluster a user refers to", opensearchRefMismatch)))
+		})
+	})
+
+	When("cluster is not ready", func() {
+		BeforeEach(func() {
+			recorder = record.NewFakeRecorder(1)
+		})
+		It("should wait for the cluster to be running", func() {
+			go func() {
+				defer GinkgoRecover()
+				defer close(recorder.Events)
+				result, err := reconciler.Reconcile()
+				Expect(err).NotTo(HaveOccurred())
+				Expect(result.Requeue).To(BeTrue())
+			}()
+			var events []string
+			for msg := range recorder.Events {
+				events = append(events, msg)
+			}
+			Expect(len(events)).To(Equal(1))
+			Expect(events[0]).To(Equal(fmt.Sprintf("Normal %s waiting for opensearch cluster status to be running", opensearchPendingReason)))
+		})
+	})
+
+	Context("cluster is ready", func() {
+		extraContextCalls := 1
+		BeforeEach(func() {
+			Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(cluster), cluster)).To(Succeed())
+			cluster.Status.Phase = opsterv1.PhaseRunning
+			cluster.Status.ComponentsStatus = []opsterv1.ComponentStatus{}
+			Expect(k8sClient.Status().Update(context.Background(), cluster)).To(Succeed())
+			Eventually(func() string {
+				err := k8sClient.Get(context.Background(), client.ObjectKeyFromObject(cluster), cluster)
+				if err != nil {
+					return "failed"
+				}
+				return cluster.Status.Phase
+			}).Should(Equal(opsterv1.PhaseRunning))
+
+			transport.RegisterResponder(
+				http.MethodGet,
+				fmt.Sprintf(
+					"https://%s.%s.svc.cluster.local:9200/",
+					cluster.Spec.General.ServiceName,
+					cluster.Namespace,
+				),
+				httpmock.NewStringResponder(200, "OK").Times(2, failMessage),
+			)
+
+			transport.RegisterResponder(
+				http.MethodHead,
+				fmt.Sprintf(
+					"https://%s.%s.svc.cluster.local:9200/",
+					cluster.Spec.General.ServiceName,
+					cluster.Namespace,
+				),
+				httpmock.NewStringResponder(200, "OK").Once(failMessage),
+			)
+		})
+
+		When("existing status is true", func() {
+			BeforeEach(func() {
+				instance.Status.ExistingRole = pointer.BoolPtr(true)
+			})
+
+			It("should do nothing", func() {
+
```

Comment:
The check doe this already because there's no additional handlers in the httpmock.  The way the mock is set up if the reconciler tries to actually call any endpoints it will error.